src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2984,11 +2991,20 @@ def set_up_network_profile(self, mc: ManagedCluster) -> ManagedCluster:
 
         advanced_network_observability = self.context.get_enable_advanced_network_observability()
         if advanced_network_observability is not None:
-            network_profile.advanced_networking = self.models.AdvancedNetworking(  # pylint: disable=no-member
-                observability=self.models.AdvancedNetworkingObservability(  # pylint: disable=no-member
-                    enabled=advanced_network_observability
+            # Create an advanced networking model with an observability model if it does not exist.
+            if network_profile.advanced_networking is None:
+                network_profile.advanced_networking = self.models.AdvancedNetworking(  # pylint: disable=no-member
```

Comment:
You can just initialize the model in the if statement, and then assign values ​​to observability outside it, so you don't have to write the same code twice

Reply:
good call updated!